ui/src/app/base/sagas/websockets.ts

```diff
@@ -337,6 +353,54 @@ export function* handleFileContextRequest({
   return !!fileContextRequest;
 }
 
+/**
+ * Run a timer to keep dispatching an action.
+ * @param action - A websocket action.
+ */
+export function* pollAction(action: WebSocketAction): SagaGenerator<void> {
+  const endpoint = `${action.meta?.model}.${action.meta?.method}`;
+  try {
+    while (true) {
+      // The delay is put first as the action will have already been handled
+      // when it is first dispatched, so this should start by waiting until the
+      // next interval.
+      yield* delay((action.meta?.pollInterval || DEFAULT_POLL_INTERVAL) * 1000);
```

Comment:
I feel like we should just work with milliseconds because that's how all JS timers work by default.

Reply:
Good idea.